lib/swaps/Swaps.ts

```diff
@@ -307,6 +341,23 @@ class Swaps extends EventEmitter {
     // TODO: calculate CLTV
     if (deal.myRole === SwapDealRole.Maker) {
 	  this.logger.debug('Executing maker code');
+
+      // Verify that taker got right amount.
+	  if (amount < deal.makerAmount * 1000) {
+    this.logger.error('received ' + amount + ' mSat, expected ' + deal.makerAmount + 'mSat');
+    this.setDealState(deal, SwapDealState.Error, 'Amount sent from Taker to Maker is too small');
+    return deal.stateReason;
+  }
+
+      // verify taker's takerCltvDelta
+      if (deal.makerCltvDelta! > resolveRequest.getTimeout() - resolveRequest.getHeightNow()) {
+        this.logger.error('got timeout ' + resolveRequest.getTimeout() + ' at height ' + resolveRequest.getHeightNow());
+        this.logger.error('cltvDelta is ' + (resolveRequest.getTimeout() - resolveRequest.getHeightNow()) +
+            'expected delta of ' + deal.makerCltvDelta);
+        this.setDealState(deal, SwapDealState.Error, 'cltvDelta sent from Taker to Maker is too small');
+        return deal.stateReason;
```

Comment:
Doesn't have to be addressed in this PR, but I feel this should be treated as an error case from the grpc perspective. Right now calling `resolveHash` will never result in a grpc error response with a code and description, even when the operation has clearly failed. From the perspective of lnd, I guess it would receive back this string message, see that it's not the preimage for the payment hash, and then fail the payment?

Reply:
Yes.  Not for this PR and highly dependent on the LND change.
I did it like this to keep the LND change smaller and less compliacted. The preImage is checked against the hash for validity on LND side so I can return whatever I wish.